src/__experimental__/components/form/form.component.js

```diff
@@ -390,22 +378,33 @@ class FormWithoutValidations extends React.Component {
 
   /** Clone the children, pass in callback to allow form to store controlled data */
   renderChildren() {
-    const { children } = this.props;
+    const { children, isLabelRightAligned } = this.props;
 
     if (!children) return null;
 
     const childrenArray = Array.isArray(children) ? children : [children];
 
-    return childrenArray.map((child) => {
-      return React.cloneElement(child, { ...child.props, addInputToFormState: this.addInputDataToState });
+    return childrenArray.map((child, index) => {
+      if (typeof child.type !== 'function') return child;
+
+      return React.cloneElement(child, {
+        ...child.props,
+        key: String(index),
```

Comment:
It's generally [not recommended](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md) to use an array index as a React key.

Reply:
Yeah this was to get see if it would remove the warnings, I'll use guids instead

Reply:
Thanks :white_check_mark: